mqtt/mqtt-broker/src/configuration.rs

```diff
@@ -0,0 +1,211 @@
+use config::{Config, ConfigError, File, FileFormat};
+use regex::Regex;
+use serde::{Deserialize, Deserializer};
+use std::path::Path;
+use std::time::Duration;
+
+pub const DEFAULTS: &str = include_str!("config.json");
+
+#[derive(Debug, Deserialize)]
+pub struct InflightMessages {
+    max_count: u32,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RetainedMessages {
+    max_count: Option<u32>,
+    #[serde(with = "humantime_serde")]
+    expiration: Option<Duration>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct SessionMessages {
+    #[serde(deserialize_with = "humansize")]
+    max_message_size: Option<u64>,
```

Comment:
Are there appropriate defaults for these sizes? for instance: `u64::MAX` This will make using this a lot less annoying because we won't need to unwrap the option.

Reply:
For some of these fields we may want to look at making them `usize`. For instance, max message size seems like a good candidate, because a 32 bit platform couldn't allocate more than usize bytes.